### ISSUE TITLE
2021-5-13

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#542]
381Set four parts for 3_message_passing; done; try to train 3 on GPU; we set g(data) on GPU and set model on GPU, then it is OK; (2021-5-7)
382AI 框架部署方案之模型部署概述 https://mp.weixin.qq.com/s/Rb_mg6EWVBbnusPuVgbD2A (2021-5-8)
383Try to run 4_link_prediction; done; Link Prediction using Graph Neural Networks; set 4 run on GPU; Overview of Link Prediction with GNN; social recommendation, item recommendation, knowledge graph completion; Loading graph and features; Prepare training and testing sets; 
384Make Your Own Dataset; DGLDataset Object Overview; 
385Interaction contain dest(node id), source(node id), and weight; members contain id, club, and age; (2021-5-8)
386Pandas library: 在计算机编程中，pandas是Python编程语言的用于数据操纵和分析的软件库。特别是，它提供操纵数值表格和时间序列的数据结构和运算操作。它是在三条款BSD许可证下发行的自由软件[2]。它的名字衍生自术语“面板数据”（panel data），这是计量经济学的数据集术语，它们包括了对同一个体的在多个时期上的观测[3]。它的名字是短语“Python data analysis”自身的文字游戏[4]。(2021-5-8)
387Print members and members.head() to see what the fuck they are; what are function pd.read_csv? What is pandas head? Return the first 5 rows; (2021-5-8)
388 demonstrating how to add node features into the graph for dataset creation; strategies like binning or normalizing the feature would work better; (2021-5-8)
389What is torch.from_numpy? The returned tensor and ndarray share the same memory. Modifications to the tensor will be reflected in the ndarray and vice versa. The returned tensor is not resizable. (2021-5-8)
